state/factory/workingset.go

```diff
@@ -166,6 +167,14 @@ func (ws *workingSet) runAction(
 	if !ok {
 		return nil, errors.New("protocol is empty")
 	}
+	// if it's an EVM tx container, unfold the tx inside
+	if protocol.MustGetFeatureCtx(ctx).UseEvmContainer {
+		if container, ok := selp.Action().(action.EvmTxContainer); ok {
+			if err := container.Unfold(selp, ctx, ws.checkContract); err != nil {
+				return nil, errors.Wrapf(err, "Failed to unfold EVM tx inside the container")
```

Comment:
if an error is returned for a container with invalid tx, `pickAndRunActions` will quit. Is this an expected behavior?

Reply:
1. it is not expected
2. this applies to other error returns (not a new issue caused by this tx container), added a TODO